pkg/ottl/expression_test.go

```diff
@@ -111,17 +111,19 @@ func basicSliceByte() (ExprFunc[any], error) {
 
 func Test_newGetter(t *testing.T) {
 	tests := []struct {
-		name string
-		val  value
-		ctx  any
-		want any
+		name        string
+		val         value
+		ctx         any
+		want        any
+		wantLiteral bool
 	}{
 		{
 			name: "string literal",
 			val: value{
 				String: ottltest.Strp("str"),
 			},
-			want: "str",
+			want:        "str",
+			wantLiteral: true,
```

Comment:
Could you please add some extra conditions here (or maybe specific tests for `listGetter` & `mapGetter`) to cover literal getters holding non-literal values, for example, a slice holding non literal values: `["foo", span.name]`, or a map holding paths/expressions `{"key": "foo", "value": ToUpperCase("bar")}` ?

Reply:
This should already be covered by the test cases added in `Test_NewFunctionCall`, e.g. this one: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40924/files#diff-3f8822d6a24dc5a9d3a52c3c72fdf45f1621e52d5e782b8576401cca2a913330R1167